pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_base_instrument.py

```diff
@@ -98,99 +760,493 @@ def add_parameters_from_file(self, filename: str):
-                    vals=vals.Ints(int(par[2]), int(par[3])))
+    def _create_parameter_file(self, filename: str):
+        """
+        This generates a json file Containing the node_docs as extracted
+        from the ZI instrument API.
+
+        Replaces the use of the s_node_pars and d_node_pars files.
+        """
+        # Get all interesting nodes
+        nodes = json.loads(self.daq.listNodesJSON('/' + self.devname))
+        
+        modified_nodes = {}
+       
+        # Do some name mangling
+        for name, node in nodes.items():
+            name = name.replace('/' + self.devname.upper() + '/', '')
+            node['Node'] = name
+            modified_nodes[name] = node
+
+        # Dump the nodes
+        with open(filename, "w") as json_file:
+            json.dump(modified_nodes, json_file, indent=4, sort_keys=True)
+
+    def _is_device_connected(self, device):
+        """
+        Return true if the given device is already connected to the server.
+        """
+        if device.lower() in [x.lower() for x in self.daq.getString('/zi/devices/connected').split(',')]:
+            return True
+        else:
+            return False
+
+    def _get_full_path(self, paths):
+        """
+        Concatenates the device name with one or more paths to create a fully
+        qualified path for use in the server.
+        """
+        if type(paths) is list:
+            for p, n in enumerate(paths):
+                if p[0] != '/':
+                    paths[n] = ('/' + self.devname + '/' + p).lower()
+                else:
+                    paths[n] = paths[n].lower()
+        else:
+            if paths[0] != '/':
+                paths = ('/' + self.devname + '/' + paths).lower()
             else:
-                print("parameter {} type {} from from"
-                      " s_node_pars not recognized".format(parname, par[1]))
+                paths = paths.lower()
+
+        return paths
 
-    @classmethod
-    def _gen_set_func(self, dev_set_type, node_path: str):
+    def _get_awg_directory(self):
+        return os.path.join(self._awgModule.get('awgModule/directory')['directory'][0], 'awg')
+
+    def _initialize_waveform_to_zeros(self):
         """
-        Generates a set function based on the dev_set_type method (e.g., seti)
-        and the node_path (e.g., '/dev8003/sigouts/1/mode'
+        Generates all zeros waveforms for all codewords.
         """
-        def set_func(val):
-            dev_set_type(node_path, val)
-            return dev_set_type(node_path, value=val)
-        return set_func
+        t0 = time.time()
+        wf = numpy.zeros(self._default_waveform_length)
+        waveform_params = [value for key, value in self.parameters.items()
+                           if 'wave_ch' in key.lower()]
+        for par in waveform_params:
+            par(wf)
+        t1 = time.time()
+        print('Set all waveforms to zeros in {:.1f} ms'.format(1.0e3*(t1-t0)))
```

Comment:
I'd prefer log.debug because that allows filtering

Reply:
Fixed.